src/store-api/src/metadata.rs

```diff
@@ -45,6 +47,32 @@ pub struct ColumnMetadata {
     pub column_id: ColumnId,
 }
 
+impl ColumnMetadata {
+    /// Construct `Self` from protobuf struct [ColumnDef]
+    pub fn from_column_def(column_def: ColumnDef) -> Result<Self> {
```

Comment:
question(non-blocking): Should we implement `TryFrom`?

Reply:
I prefer not to. It's hard to find